### PR TITLE
[fish] Fix ctrl-r regression in versions <2.4

### PR DESCRIPTION
**Pull request by tomfitzhenry**
... but multiline won't be supported in versions <2.4

Issue #966

I have yet to test this on version 2.3, but will.

### CODE REVIEW

File: shell/key-bindings.fish [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Key bindings
 # ------------
 function fzf_key_bindings
-
+  set -g __FZF_FISH_MAJOR (echo $FISH_VERSION | cut -f1 -d.)
```
Comment (tomfitzhenry) [inline shell/key-bindings.fish]: This creates a global variable, which isn't ideal. Ideally we'd be able to set a variable that fzf-history-widget has access to, but so far I've been unable to find a way.

**Comment by tomfitzhenry** [comment]
I reproduced the bug on Debian Jessie with fish 2.2.0 and fzf 0.16.8, and can confirm that this PR fixes the issue.

File: shell/key-bindings.fish (hunk 1 above)
Comment (junegunn) [inline shell/key-bindings.fish]: How about just doing this inside `fzf-history-widget`? We can create local variables instead only when they're needed, which makes us repeat it whenever the widget's called, but it won't cause any noticeable delay.

Comment (tomfitzhenry) [inline shell/key-bindings.fish]: Ok. I had expected people would prefer a global variable to unnecessary computation, but I'm happen with either.

Comment (junegunn) [inline shell/key-bindings.fish]: I'm fine with either actually. I'm not even a fish user 😆 

Maybe fish has some string manipulation functions we can use in this case, so we don't have to start two subprocesses?

Comment (tomfitzhenry) [inline shell/key-bindings.fish]: According to Wikipedia, fish is an "exotic" shell!

Fish does have string manipulation functions, but only as of 2.3 (see https://fishshell.com/docs/current/commands.html#string ). I figured supporting fish <2.3 was more important saving subprocess invocations.

File: shell/key-bindings.fish [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Key bindings
 # ------------
 function fzf_key_bindings
-
+
```
Comment (junegunn) [inline shell/key-bindings.fish]: Everything looks good except this line :)

Comment (tomfitzhenry) [inline shell/key-bindings.fish]: Fixed!

File: shell/key-bindings.fish [hunk 3]
```diff
@@ -39,8 +39,20 @@ function fzf_key_bindings
...
+      # history's -z flag is needed for multi-line support.
+      # history's -z flag was added in fish 2.4.0, so don't use it for versions
+      # before 2.4.0.
+      if [ "$FISH_MAJOR" -gt 2 -o \( "$FISH_MAJOR" -eq 2 -a "$FISH_MINOR" -ge 4 \) ];
+          history -z | eval (__fzfcmd) --read0 -q '(commandline)' | perl -pe 'chomp if eof' | read -lz result
+          and commandline -- $result
+      else
+          history | eval (__fzfcmd) -q '(commandline)' | read -l result
+          and commandline -- $result
```
Comment (junegunn) [inline shell/key-bindings.fish]: I'm sorry, but I just realized that you used 4-space indentation instead of 2. Can you fix it as well?

**Comment by junegunn** [comment]
Thanks!

**Pull request merged**